File: backend spring/src/main/java/com/bloodbank/entity/BloodRequest.java
package com.bloodbank.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "blood_requests")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class BloodRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "request_id")
    private Long requestId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_user_id", nullable = false)
    private User hospital;

    @Column(name = "blood_group", nullable = false, length = 5)
    private String bloodGroup;

    @Column(name = "units_requested", nullable = false)
    private int unitsRequested;

    @Column(nullable = false, length = 20)
    private String urgency;

    @Column(nullable = false, length = 20)
    private String status;

    // Extra Columns from our DB design
    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @Column(name = "is_fulfilled", nullable = false)
    private boolean isFulfilled = false;
}

File: BackendDotNet/Models/BloodRequest.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackendDotNet.Models
{
    [Table("blood_requests")]
    public class BloodRequest
    {
        [Key]
        [Column("request_id")]
        public long RequestId { get; set; }

        [Required]
        [Column("hospital_user_id")]
        public long HospitalUserId { get; set; }

        // Alias for compatibility with repository methods
        public long HospitalId => HospitalUserId;

        [Required]
        [Column("blood_group")]
        [StringLength(5)]
        public string BloodGroup { get; set; } = string.Empty;

        [Required]
        [Column("units_requested")]
        public int UnitsRequested { get; set; }

        [Required]
        [Column("urgency")]
        [StringLength(20)]
        public string UrgencyLevel { get; set; } = string.Empty;

        [Required]
        [Column("status")]
        [StringLength(20)]
        public string Status { get; set; } = string.Empty;

        [Column("request_date")]
        public DateTime RequestDate { get; set; } = DateTime.UtcNow;

        [Column("required_date")]
        public DateTime RequiredDate { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [Column("is_fulfilled")]
        public bool IsFulfilled { get; set; } = false;

        // Navigation property
        [ForeignKey("HospitalUserId")]
        public virtual User Hospital { get; set; } = null!;
    }
}
